### PR TITLE
GenerateConfig will a prompt to overwrite when config already exists

### LINKED ISSUE #2: GenerateConfig should show a prompt when config already exists

### PR DESCRIPTION
**Pull request by samarthkathal**
- GenerateConfig will a prompt to overwrite when config already exists.

Resolves #2

### CODE REVIEW

File: lib/rubocop_director/commands/generate_config.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ class GenerateConfig
       include Dry::Monads::Do.for(:run)
 
       RUBOCOP_TODO = ".rubocop_todo.yml"
+      RUBOCOP_DIRECTOR_CONFIG =  ".rubocop-director.yml"
```
Comment (DmitryTsepelev) [inline lib/rubocop_director/commands/generate_config.rb]: We have it here already https://github.com/DmitryTsepelev/rubocop_director/blob/master/lib/rubocop_director.rb#L7

File: lib/rubocop_director/commands/generate_config.rb [hunk 2]
```diff
@@ -28,6 +28,8 @@ def run
       private
 
       def load_config
+        return Failure("#{RUBOCOP_DIRECTOR_CONFIG} already exists") if File.file?(RUBOCOP_DIRECTOR_CONFIG)
```
Comment (DmitryTsepelev) [inline lib/rubocop_director/commands/generate_config.rb]: I thought we will _ask_ user if he wants to override file, i.e.

```ruby
puts "#{RUBOCOP_DIRECTOR_CONFIG} already exists, do you want to override it? (y, n)"
if gets == 'y' 
```

**Review by DmitryTsepelev — CHANGES_REQUESTED** [review]

File: lib/rubocop_director/commands/generate_config.rb (hunk 1 above)
Comment (samarthkathal) [reply lib/rubocop_director/commands/generate_config.rb]: Thanks

File: lib/rubocop_director/commands/generate_config.rb (hunk 2 above)
Comment (samarthkathal) [reply lib/rubocop_director/commands/generate_config.rb]: Okay, that makes sense.

**Comment by samarthkathal** [comment]
@DmitryTsepelev I've made the necessary changes now. Should we also try and show the `configs` diff for the case when user wants to overwrite the previously generated configs?

File: lib/rubocop_director/commands/generate_config.rb [hunk 3]
```diff
@@ -28,6 +27,12 @@ def run
       private
 
       def load_config
+        if File.file?(CONFIG_NAME)
+          puts("#{CONFIG_NAME} already exists, do you want to override it? (y, n)")
+
+          return Failure("previous version of #{CONFIG_NAME} was preserved.") unless $stdin.gets.chomp == "y"
```
Comment (DmitryTsepelev) [inline lib/rubocop_director/commands/generate_config.rb]: I feel like it should be success — user got what he wanted 🙂

**Review by DmitryTsepelev — COMMENTED** [review]
Almost there! I added one question below and we're good to go 🙂 I do not think we need a diff right now — I have a plan to add another command like "show me cops that are in TODO but have no weights", it should cover most of cases when user already has config and wants to update it

File: lib/rubocop_director/commands/generate_config.rb (hunk 3 above)
Comment (samarthkathal) [reply lib/rubocop_director/commands/generate_config.rb]: Makes sense. I went with a failure result as I did not want to introduce another condition [here](https://github.com/samarthkathal/rubocop_director/blob/758ac8581a4e9d4b1224c9db6c50cfae079722e0/lib/rubocop_director/commands/generate_config.rb#L12).

I am not sure how I would do that with monads. do you have any suggestions?

**Comment by samarthkathal** [comment]
I have refactored the code to use truthy functions now. There may be a way to use monads here as well but I am not sure how to go about it.

File: lib/rubocop_director/commands/generate_config.rb (hunk 3 above)
Comment (DmitryTsepelev) [reply lib/rubocop_director/commands/generate_config.rb]: good point, failure was better, I'll merge it in and try to fix 🙂thank you so much!

**Pull request merged**